pilot/pkg/networking/core/v1alpha3/cluster_cache.go

```diff
@@ -164,6 +179,7 @@ func (c cacheStats) merge(other cacheStats) cacheStats {
 
 func buildClusterKey(service *model.Service, port *model.Port, cb *ClusterBuilder, proxy *model.Proxy, efKeys []string) *clusterCache {
 	clusterName := model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, port.Port)
+	// TODO if this is a non EDS cluster take endpoint builder keys into account
```

Comment:
So just taking the EB key may not be enough. EDS has a bunch of hacks to account for the fact that EDS updates dynamically; https://github.com/istio/istio/blob/d0d87b09f633d940fac8203ac7ade953db0416b8/pilot/pkg/xds/eds.go#L182 for example.

I suppose the same can apply to CDS as well as long as this TODO is done and DependentConfigs is correct (I think its fine already, but need to check).

@ramaraochavali @hzxuzhonghu is there anything else that may break from the broken relationship between PushContext => XDS output being stable? Nothing comes to mind immediately.

Reply:
The service resolution type is DNS here, so i think it is simpler